### PR TITLE
Sanitize text

### LINKED ISSUE #573: Replace `innerHTML` by safer and more performant methods
https://github.com/mozilla/firefox-translations/blob/7c7f4fc667eee7b04b8e9a7ec85b2fb3e5501461/extension/view/popup/js/index.js#L82 and https://github.com/mozilla/firefox-translations/blob/7c7f4fc667eee7b04b8e9a7ec85b2fb3e5501461/extension/view/popup/js/index.js#L87 should be refactored into [safe DOM manipulation](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Safely_inserting_external_content_into_a_page) that does not evaluate strings as code.

### PR DESCRIPTION
**Pull request by andrenatal**
fixes #573

### CODE REVIEW

File: extension/view/popup/js/index.js [hunk 1]
```diff
@@ -79,12 +79,12 @@ class Mediator {
 }
 
 const setLangs = (selector, langsToSet, value, exclude) => {
-    selector.innerHTML = `<option value="0">${browser.i18n.getMessage("languageDefaultOption")}</option>`;
+    selector.innerHTML = DOMPurify.sanitize(`<option value="0">${browser.i18n.getMessage("languageDefaultOption")}</option>`, { USE_PROFILES: { html: true } });
```
Comment (Rob--W) [inline extension/view/popup/js/index.js:82]:
```suggestion
    selector.textContent = "";
    selector.add(new Option(browser.i18n.getMessage("languageDefaultOption"), "0"));
```

File: extension/view/popup/js/index.js [hunk 2]
```diff
@@ -79,12 +79,12 @@ class Mediator {
...
     for (const [code, type] of Object.entries(langsToSet)) {
         if (code !== exclude) {
             let name = langs.get(code);
             if (type === "dev") name += " (Beta)";
-            selector.innerHTML += `<option value="${code}">${name}</option>`;
+            selector.innerHTML += DOMPurify.sanitize(`<option value="${code}">${name}</option>`, { USE_PROFILES: { html: true } });
```
Comment (Rob--W) [inline extension/view/popup/js/index.js:87]:
```suggestion
            selector.add(new Option(name, code));
```

File: extension/view/popup/index.html [hunk 3]
```diff
@@ -26,6 +26,7 @@
       </div>
       <div class="footer" id="status"></div>
     </div>
+    <script src="/model/static/DOMPurify/purify.min.js"></script>
```
Comment (Rob--W) [inline extension/view/popup/index.html:29]:
```suggestion
```

**Review by Rob--W — CHANGES_REQUESTED** [review]
The proposed use of DOMPurify is a bad fit. I suggest to use the `Option` constructor instead, to achieve exactly the same behavior in an efficient and readable way.

Relevant documentation:
- https://developer.mozilla.org/en-US/docs/Web/API/HTMLOptionElement/Option
- https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/add

**Comment by andrenatal** [comment]
> The proposed use of DOMPurify is a bad fit. I suggest to use the `Option` constructor instead, to achieve exactly the same behavior in an efficient and readable way.
> 
> Relevant documentation:
> 
>     * https://developer.mozilla.org/en-US/docs/Web/API/HTMLOptionElement/Option
> 
>     * https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/add

@Rob--W ok maybe is better to then update this which was recommended in the PR https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Safely_inserting_external_content_into_a_page#html_sanitization

**Review by Rob--W — APPROVED** [review]

**Pull request merged**